src/main/scala/fi/oph/koski/mydata/ApiProxyServlet.scala

```diff
@@ -0,0 +1,50 @@
+package fi.oph.koski.mydata
+
+import fi.oph.koski.config.KoskiApplication
+import fi.oph.koski.db.GlobalExecutionContext
+import fi.oph.koski.http.KoskiErrorCategory
+import fi.oph.koski.log.Logging
+import fi.oph.koski.opiskeluoikeus.OpiskeluoikeusQueries
+import fi.oph.koski.servlet.{ApiServlet, InvalidRequestException, NoCache}
+import org.scalatra.ContentEncodingSupport
+import scala.collection.JavaConverters._
+
+
+class ApiProxyServlet(implicit val application: KoskiApplication) extends ApiServlet with Logging with GlobalExecutionContext with OpiskeluoikeusQueries with ContentEncodingSupport with NoCache {
+
+  get("/:oid") {
+    def studentId = params("oid")
+    request.header("X-ROAD-MEMBER") match {
+      case Some(memberCode) => {
+        logger.info(s"Requesting MyData content for user ${studentId} by client ${memberCode}")
+
+        def member = application.config.getConfigList("mydata.members").asScala.find(member =>
+          member.getStringList("membercodes").contains(memberCode)
+        )
+
+        member match {
+          case Some(memberConf) => {
+            def memberId = memberConf.getString("id")
```

Comment:
`def` on metodeita varten, arvoja varten suositaan `val`:in käyttöä.
Sit ehkä vähän makuasia, mutta koodia vois ehkä pilkkoa palasiin ja idiomaattisempaa scalaa tulisi jos vähemmän pattern matchailisi ja enemmän mappailisi, tyyliin:
```
  get("/:oid") {
    requestDispatcher(request.header("X-ROAD-MEMBER"), params("oid")) match {
      case Right(dispatcher) => dispatcher.forward(request, response)
      case Left(status) =>
        logger.warn(status.errorString.mkString)
        throw InvalidRequestException(status)
    }
  }

  private def requestDispatcher(xRoadHeader: Option[String], studentId: String) = xRoadHeader
      .toRight(KoskiErrorCategory.badRequest.header.missingXRoadHeader(s"Missing X-ROAD-MEMBER header when requesting student data for $studentId"))
      .flatMap(findMemberConf)
      .flatMap { memberConf =>
        val memberId = memberConf.getString("id")
        if (application.mydataService.hasAuthorizedMember(studentId, memberId)) {
          Right(servletContext.getRequestDispatcher("/api/oppija"))
        } else {
          Left(KoskiErrorCategory.badRequest.header.unauthorizedXRoadHeader(s"Student $studentId has not authorized $memberId to access their student data"))
        }
      }

  private def findMemberConf(memberCode: String) =
    application.config.getConfigList("mydata.members").asScala.find(member =>
      member.getStringList("membercodes").contains(memberCode)
    ).toRight(KoskiErrorCategory.badRequest.header.invalidXRoadHeader(s"Unknown X-ROAD-MEMBER $memberCode when requesting student data for ${params("oid")}"))
```